src/mmds/src/ns.rs

```diff
@@ -119,17 +119,22 @@ impl MmdsNetworkStack {
     // This is the entry point into the MMDS network stack. The src slice should hold the contents
     // of an Ethernet frame (of that exact size, without the CRC).
     pub fn detour_frame(&mut self, src: &[u8]) -> bool {
-        // The frame cannot possibly contain an ARP request or IPv4 packet for the MMDS.
-        if !test_speculative_tpa(src, self.ipv4_addr)
-            && !test_speculative_dst_addr(src, self.ipv4_addr)
-        {
-            return false;
-        }
-
         if let Ok(eth) = EthernetFrame::from_bytes(src) {
             match eth.ethertype() {
-                ETHERTYPE_ARP => return self.detour_arp(eth),
-                ETHERTYPE_IPV4 => return self.detour_ipv4(eth),
+                ETHERTYPE_ARP => {
+                    if !test_speculative_tpa(src, self.ipv4_addr) {
+                        METRICS.mmds.rx_accepted_unusual.inc();
+                        return false;
+                    }
+                    return self.detour_arp(eth);
+                }
+                ETHERTYPE_IPV4 => {
+                    if !test_speculative_dst_addr(src, self.ipv4_addr) {
+                        METRICS.mmds.rx_accepted_unusual.inc();
```

Comment:
Won't this metric be incremented for all IPv4 packets _not destined_ to mmds?

I thought `detour_frame()` is called for all frames on an mmds-enabled iface. The function then checks whether frame is destined for MMDS based on dst addr, and returns `false` otherwise (aka non-MMDS packet/frame). In which case this code increments `mmds.rx_accepted_unusual` metric for every non-mmds frame on an mmds-enabled interface.

Not sure what the purpose of this metric is, but the name is definitely disconnected from its behavior in this case.

Reply:
indeed, you are right.